test/puppetlabs/http/client/async_plaintext_test.clj

```diff
@@ -433,3 +440,15 @@
              (with-open [client (async/create-client {})]
                (let [response @(common/get client url {:as :text})]
                  (is (= 200 (:status response)))))))))))
+
+(deftest accept-language-async
+  (testing "client passes on the user-locale in Accept-Language header"
+    (testlogging/with-test-logging
+      (testwebserver/with-test-webserver app-with-language-header-echo port
+        (binding [i18n/*locale* (Locale. "es" "ES")]
```

Comment:
Would this work if you changed it to this instead?

```clj
(i18n/with-user-locale (Locale. "es" "ES") ...
```

If so, might be better to do it this way in order to avoid leaking knowledge about how the binding is done outside of the i18n library, maybe making this more future-proof.